File: 프로그래머스/unrated/181912. 배열 만들기 5/배열 만들기 5.java

import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(String[] intStrs, int k, int s, int l) {
        List<Integer> answer = new ArrayList<>();
        
        for (String intStr : intStrs) {
            int num = Integer.valueOf(intStr.substring(s, s+l));
            if (k < num) answer.add(num);
        }
        
        return answer.stream().mapToInt(e -> e).toArray();
    }
}

File: 프로그래머스/unrated/181912. 배열 만들기 5/배열 만들기 5.py
def solution(intStrs, k, s, l):
    answer = []
    
    for str in intStrs:
        if k < int(str[s:s+l]):
            answer.append(int(str[s:s+l]))
            
    return [int(str[s:s+l]) for str in intStrs if k < int(str[s:s+l])]